test/SubmitUpgradeProposal.t.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.26;
+
+import {Test, console2} from "forge-std/Test.sol";
+import {L2ArbitrumGovernorV2Test} from "test/L2ArbitrumGovernorV2.t.sol";
+import {SubmitUpgradeProposal} from "script/SubmitUpgradeProposal.s.sol";
+import {TimelockControllerUpgradeable} from "openzeppelin-upgradeable/governance/TimelockControllerUpgradeable.sol";
+import {GovernorUpgradeable} from "openzeppelin-upgradeable/governance/GovernorUpgradeable.sol";
+import {IGovernor} from "openzeppelin-contracts/contracts/governance/IGovernor.sol";
+import {BaseGovernorDeployer} from "script/BaseGovernorDeployer.sol";
+import {DeployCoreGovernor} from "script/DeployCoreGovernor.s.sol";
+import {DeployTreasuryGovernor} from "script/DeployTreasuryGovernor.s.sol";
+import {TimelockRolesUpgrader} from "src/gov-action-contracts/TimelockRolesUpgrader.sol";
+import {L2ArbitrumGovernorV2} from "src/L2ArbitrumGovernorV2.sol";
+import {SharedGovernorConstants} from "script/SharedGovernorConstants.sol";
+import {DeployImplementation} from "script/DeployImplementation.s.sol";
+
+contract SubmitUpgradeProposalTest is SharedGovernorConstants, Test {
+  uint256 constant FORK_BLOCK = 220_819_857; // Arbitrary recent block
+  /// @dev Proxy admin contract deployed in construction of TransparentUpgradeableProxy -- getter is internal, so we
+  /// hardcode the address below
+  address constant PROXY_ADMIN_CONTRACT = 0x740f24A3cbF1fbA1226C6018511F96d1055ce961; // Proxy Admin Contract Address
+
+  SubmitUpgradeProposal submitUpgradeProposal;
```

Comment:
nit: this seems a little clearer

```suggestion
  SubmitUpgradeProposalScript submitUpgradeProposalScript;
```

Reply:
https://github.com/ScopeLift/arbitrum-governor-upgrade/pull/11/commits/509970dbcd8bd3dbbda222f38b8721852a99d4b2